provider.go

```diff
@@ -0,0 +1,38 @@
+package summaraizer
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"text/template"
+)
+
+// AiProvider is an interface that defines an AI provider to summarize comments.
+type AiProvider interface {
+	Summarize(reader io.Reader) (string, error)
+}
+
+// Common is a generic struct that shares common fields between all AI providers.
+type Common struct {
```

Comment:
Too much abstraction at this point. Having a few duplicate fields better than wrong abstraction so early in a project may require huge refactoring to all affected entities later. So better have some duplicate fields in AI providers.

Reply:
Thanks!
Fixed here https://github.com/ioki-mobility/summaraizer/pull/23/commits/fe89d6a6395b2c3268dac52cbc3d804d5bda3ea9